salt/hosts-minion/init.sls

```diff
@@ -0,0 +1,14 @@
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
+{{ minion_id }}-host-entry:
+  host.present:
+{% if addrlist is string %}
+    - ip: {{ addrlist }}
+{% else %}
+    - ip: {{ addrlist|first }}
+{% endif %}
+    - names:
+      - api
+      - api.{{ pillar['internal_infra_domain'] }}
+      - {{ minion_id }}
+      - {{ minion_id }}.{{ pillar['internal_infra_domain'] }}
+{% endfor %}
```

Comment:
I think we can improve this. We will access the api servers by the `api` alias and never by the hostname (so we don't care about adding entries like `<hash>.<domain>`) and, at the same time, it would be good to point `api.infra.caasp.local` to one of the API servers, but choosing a random one between all the servers available.

I think we can achieve these things with:

```
{% set api_servers = salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').values() %}

api-host-entry:
  host.present:
{% if api_servers is string %}
    - ip: {{ api_servers }}
{% else %}
    - ip: {{ api_servers|first }}
{% endif %}
    - names:
      - api
      - api.{{ pillar['internal_infra_domain'] }}
{% endfor %}```


Lists in Jinja2 are _supposed_ to be _unordered_, so `api_servers|first` should be a random API server...